app/src/main/java/atlant/moviesapp/presenters/MovieListPresenter.java

```diff
@@ -0,0 +1,73 @@
+package atlant.moviesapp.presenters;
+
+import android.util.Log;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import atlant.moviesapp.model.Movie;
+import atlant.moviesapp.model.MovieGenre;
+import atlant.moviesapp.model.MoviesResponse;
+import atlant.moviesapp.rest.ApiClient;
+import atlant.moviesapp.rest.ApiInterface;
+import atlant.moviesapp.views.MovieListView;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+/**
+ * Created by Korisnik on 12.04.2017..
+ */
+
+public class MovieListPresenter {
+
+    private MovieListView view;
+    private static final int MOST_POPULAR = 0;
+    private static final int LATEST = 1;
+    private static final int HIGHEST_RATED = 2;
+
+
+    private final static String API_KEY = "e87991d039ddbe98de1958bf8a8b5148";
+
+    public MovieListPresenter(MovieListView view) {
+        this.view = view;
+    }
+
+    public void getHighestRatedMovies(int tag) {
```

Comment:
Make the call a private variable of the MovieListPresenter class, And cancel a call when needed. 
Read the following link : https://futurestud.io/tutorials/retrofit-2-cancel-requests
Make the presenter have some of the lifecycle methods which the fragment has, and call them from the fragment in order the release memory, or cancel requests if needed.